test/testwebsocket.cpp

```diff
@@ -0,0 +1,102 @@
+#include <QTest>
+#include <chrono>
+#include <thread>
+
+#include "websocket.h"
+#include "common.h"
+
+using namespace std::chrono_literals;
+
+QSharedPointer<OCC::WebSocket>
+createWebSocket()
+{
+    return QSharedPointer<OCC::WebSocket>(new OCC::WebSocket);
+}
+
+class TestWebSocket : public QObject
```

Comment:
Your pretty much testing Qt signal/slot system here. Yes, that also check you didn't screw up your connects... still this makes for a fairly low value test. Reaching out further in the stack in your other test suite would bring more value.

Reply:
Yes, I know. I just wanted to get sure because I have not worked that much with Qt.